packages/nimble-components/src/select/index.ts

```diff
@@ -967,6 +1091,8 @@ export class Select
             return;
         }
 
+        this.indexWhenOpened = this.selectedIndex;
```

Comment:
This comment is for a few lines above, but it won't let me comment there since the code isn't changed.

What is the purpose of the `if (!this.open)` block within `initializeOpenState()`? Does this function get called when the select is closing? I ask because I noticed some duplicate logic between this if-statement and `openChanged`, and I'm not sure if that's intentional or not. Does `this.openActiveIndex = undefined;` need to be in both places as well, or is the duplication unnecessary?